File: scalademo/src/com/milko/trainings/scala/interfaces/scalac/Customers.scala
package com.milko.trainings.scala.interfaces.scalac

import com.milko.trainings.scala.inheritance.scalac.Customer
import scala.collection.mutable.HashSet
import com.milko.trainings.scala.inheritance.scalac.DiscountedCustome

class Customers extends Sortable[Customer] {
  
  private val customers = scala.collection.mutable.Set[Customer]();
  
  def add(customer: Customer) = customers.add(customer);
  
  def iterator: Iterator[Customer] = customers.iterator;
}

object Customers {
  def main(args: Array[String]): Unit = {
    val customers = new Customers();
    customers.add(new Customer("Milko Galev", "Address 1"));
    customers.add(new Customer("Nora Galeva", "Address 2"));
    customers.add(new Customer("Pavel Galev", "Address 3"));
    customers.add(new Customer("Nataliya Galeva", "Address 4"));
    customers.add(new DiscountedCustome("Petar Galev", "Address 5"));
    
    println(customers.sort)
  }
}

File: scalademo/src/com/milko/trainings/scala/interfaces/javac/Customers.java
package com.milko.trainings.scala.interfaces.javac;

import java.util.HashSet;
import java.util.Iterator;
import java.util.Set;

import com.milko.trainings.scala.inheritance.javac.Customer;



public class Customers implements Sortable<Customer> {
	
	private final Set<Customer> customers = new HashSet();
	
	public void add(Customer cusmomer) {
		customers.add(cusmomer);
	}

	@Override
	public Iterator<Customer> iterator() {
		return customers.iterator();
	}

}

